add: ash tendril has heal aura that heals ash walkers

UberHelmsman opened the pull request:
<!-- Пишите **НИЖЕ** заголовков и **ВЫШЕ** комментариев, иначе ваш текст может не отобразиться. -->
<!-- В Contributing.MD вы можете найти некоторые рекомендации к оформлению пулл-реквеста. -->

## Описание
Теперь тендрил эшей хилит эшей и шамана на 0.4 брута, бёрна, токса и крови. Сделано с помощью нового трейта TRAIT_HEALS_FROM_ASH_TENDRIL, который добавлен расе эшей и шаману. Не убирал simple_heal, ибо никому не должно сделать хуже.
<!-- Опишите, что делает ваш ПР. Документировать каждую деталь не требуется, просто укажите основные изменения. -->

## Ссылка на предложение/Причина создания ПР
https://discord.com/channels/617003227182792704/1248111144619474945/1250387712410914837
Пока предложка в процессе прохождения, делаю ПР, потом отредачу по прохождению.

Сделал ПР в основном для того, чтоб заранее посмотрели и сказали, если что не так. На локалке норм работает. 
<!-- Здесь оставьте ссылку на сообщение в #отчеты-по-предложениям, чтобы подтвердить, что ваше предложение одобрено. Либо укажите, почему этот ПР должен пройти без предложки. -->
<!-- Пример ссылки: https://discord.com/channels/617003227182792704/755125334097133628/ID-сообщения -->

## Демонстрация изменений
<!-- Здесь вы можете показать изменения внешне, к примеру новые спрайты, звуки или изменения карты. Или написать, что именно изменилось для игроков. Этот пункт полностью опционален и его можно удалить. -->

UberHelmsman committed:
ash-tendril-heals-ashwalkers

ash tendril now heals ash walkers and shaman within 4 tiles of range (inside the tendril room) by 0.4 brute, burn, tox and blood damage.
In order to make it work new trait was added TRAIT_HEALS_FROM_ASH_TENDRIL.

Gottfrei commented on code/_globalvars/traits.dm:
```diff
@@ -37,6 +37,7 @@ GLOBAL_LIST_INIT(traits_by_type, list(
 		"TRAIT_HEALS_FROM_CARP_RIFTS" = TRAIT_HEALS_FROM_CARP_RIFTS,
 		"TRAIT_HEALS_FROM_CULT_PYLONS" = TRAIT_HEALS_FROM_CULT_PYLONS,
 		"TRAIT_HEALS_FROM_HOLY_PYLONS" = TRAIT_HEALS_FROM_HOLY_PYLONS,
+		"TRAIT_HEALS_FROM_ASH_TENDRIL" = TRAIT_HEALS_FROM_ASH_TENDRIL,
```
читаем что написано вверху файла.

UberHelmsman replied:
перенес трейт в соответствии с алфавитным порядком

Gottfrei replied:
не перенёс, он должен быть на строке 38.

UberHelmsman replied:
ну я просто перенес, но коммит не сделал)

UberHelmsman replied:
аутдатед

Gottfrei commented on code/modules/ruins/lavalandruin_code/ash_walker_den.dm:
```diff
@@ -94,6 +107,7 @@
 /obj/effect/mob_spawn/human/ash_walker/special(mob/living/carbon/human/new_spawn)
 	new_spawn.rename_character(new_spawn.real_name, new_spawn.dna.species.get_random_name(new_spawn.gender))
 	new_spawn.faction += "ashwalker"
+	ADD_TRAIT(new_spawn, TRAIT_HEALS_FROM_ASH_TENDRIL, INNATE_TRAIT)
```
это точно должно быть привязано здесь, а не к эшволкерам как расе? и соответственно забираться и отниматься при изменении расы?

UberHelmsman replied:
а в файле с расами трейт пишется в species_traits же?

Gottfrei replied:
нет, пока нужно выдавать вручную, для примера смотри нуклеацию.

UberHelmsman replied:
нашел думаю что надо, щас коммит сделаю

UberHelmsman replied:
аутдейтед

UberHelmsman committed:
Merge branch 'master220' into ash-tendril-heasl-ash-walkers-and-shaman

UberHelmsman committed:
трейт на хил привязывается к расе, а не спавну

UberHelmsman committed:
Merge branch 'ash-tendril-heasl-ash-walkers-and-shaman' of https://github.com/UberHelmsman/ParadiseButMine into ash-tendril-heasl-ash-walkers-and-shaman

UberHelmsman commented:
пиздец все сломалось

UberHelmsman committed:
Update unathi.dm

UberHelmsman commented:
класс все починилось

UberHelmsman commented:
еще бы ревью :)

Gottfrei commented on code/modules/mob/living/carbon/human/species/unathi.dm:
```diff
@@ -126,6 +126,7 @@
 		fire.Grant(H)
 	RegisterSignal(H, COMSIG_MOVABLE_Z_CHANGED, PROC_REF(speedylegs))
 	speedylegs(H)
+	ADD_TRAIT(H,TRAIT_HEALS_FROM_ASH_TENDRIL, INNATE_TRAIT)
```
```suggestion
	ADD_TRAIT(H,TRAIT_HEALS_FROM_ASH_TENDRIL, SPECIES_TRAIT)
```

UberHelmsman replied:
в новом коммите сделал

Gottfrei commented on code/modules/mob/living/carbon/human/species/unathi.dm:
```diff
@@ -171,6 +173,8 @@
 	if(!fire)
 		fire = new
 		fire.Grant(C)
+	ADD_TRAIT(C,TRAIT_HEALS_FROM_ASH_TENDRIL, INNATE_TRAIT)
```
у шамана уже не нужно добавлять/убирать, есть наследование.

UberHelmsman replied:
понял

UberHelmsman replied:
тоже в новом коммите сделал

Gottfrei commented on code/modules/mob/living/carbon/human/species/unathi.dm:
```diff
@@ -134,6 +135,7 @@
 	if(fire)
 		fire.Remove(H)
 	UnregisterSignal(H, COMSIG_MOVABLE_Z_CHANGED)
+	REMOVE_TRAIT(H, TRAIT_HEALS_FROM_ASH_TENDRIL, INNATE_TRAIT)
```
```suggestion
	REMOVE_TRAIT(H, TRAIT_HEALS_FROM_ASH_TENDRIL, SPECIES_TRAIT)
```

UberHelmsman replied:
тоже в новом коммите сделал

UberHelmsman committed:
убран трейт у шамана и заменен INNATE_TRAIT на SPECIES_TRAIT

UberHelmsman commented:
вроде всё

UberHelmsman committed:
выдача трейта при спавне в ручную

Выдает трейт хила при спавне эша

почему то в гитхабе коммент цвея не показали, а на почте показали. Как я понял это то что цвей имел ввиду???

Работает по аналогии с нуклеацией:
Paradise\code\modules\mob\living\carbon\human\species\nucleation.dm\on_species_gain()

UberHelmsman commented:
последний коммит ничего не ломает, но и я не разобрался, нужен ли он.

Gottfrei commented on code/modules/ruins/lavalandruin_code/ash_walker_den.dm:
```diff
@@ -92,9 +105,9 @@
 	var/eggtype = "пеплоходца"
 
 /obj/effect/mob_spawn/human/ash_walker/special(mob/living/carbon/human/new_spawn)
+	ADD_TRAIT(new_spawn, TRAIT_HEALS_FROM_ASH_TENDRIL, SPECIES_TRAIT)
```
удали

UberHelmsman committed:
удаление ненужного

UberHelmsman committed:
Merge branch 'master220' into ash-tendril-heasl-ash-walkers-and-shaman

UberHelmsman committed:
Merge branch 'master220' into ash-tendril-heasl-ash-walkers-and-shaman

UberHelmsman committed:
Merge branch 'master220' into ash-tendril-heasl-ash-walkers-and-shaman

UberHelmsman committed:
Merge branch 'master220' into ash-tendril-heasl-ash-walkers-and-shaman

UberHelmsman commented:
а вот мне надо обновлять ветку или лучше не стоит и оно само все обновится когда мерджить будут?

Gottfrei commented:
> а вот мне надо обновлять ветку или лучше не стоит и оно само все обновится когда мерджить будут?

предложка то прошла в итоге?

прошла, добавил
